src/comms/config_layer.cpp

```diff
@@ -1,21 +1,74 @@
 #include "config_layer.hpp"
 
 const Config* const ConfigLayer::configure(HIDLayer* comms) {
+    // attempt SD card load
+    config_SD_init(comms);
+    if (configured) return &config;
+
+    // if no config on SD, then await transmission
     // grab and process all config packets until finished
+    double prev_time = millis();
+    double delta_time = 0;
     while (!is_configured()) {
+    #ifdef CONFIG_LAYER_DEBUG
+        if (delta_time >= 2000) {
+            Serial.printf("Pinging for config packet....\n");
+            prev_time = millis();
+        }
+        delta_time = millis() - prev_time;
+    #endif
         comms->ping();
         process(comms->get_incoming_packet(), comms->get_outgoing_packet());
     }
 
     // put the data from the packets into the config object
     config.fill_data(config_packets, subsec_sizes);
 
+    // verify that config received matches ref system: if not, error out
+#ifndef CONFIG_OFF_ROBOT
+    Serial.printf("Received robot ID from config: %d\nRobot ID from ref system: %d\n", (int)config.robot_id, ref.ref_data.robot_performance.robot_ID);
+    if ((ref.ref_data.robot_performance.robot_ID % 100) != (int)config.robot_id) {
+        Serial.printf("ERROR: IDs do not match!! Check robot_id.cfg and robot settings from ref system!\n");
+        if (!CONFIG_ERR_HANDLER(CONFIG_ID_MISMATCH)) {
+            // honestly not sure what to put here. in current implementation ERR_HANDLER just while(1) loops
+            // this recursive call probably won't work? but it also shouldn't run? will change later if needed
+            return configure(comms);
+        }
+    }
+#endif
+
+// update stored config, msg if successful
+    Serial.println("Attempting to store config...");
+    if (store_config()) Serial.println("Config successfully stored in /config.pack");
+    else Serial.println("Config not successfully stored (is an SD card inserted?)");    // not a fatal error, can still return
+
     return &config;
 }
 
-void ConfigLayer::process(CommsPacket *in, CommsPacket *out) {
-    char *in_raw = in->raw;
-    char *out_raw = out->raw;
+void ConfigLayer::config_SD_init(HIDLayer* comms) {
+    // if on robot, we need to wait for ref to send robot_id
+    Serial.println("Waiting for ref system to initialize...");
+    while (ref.ref_data.robot_performance.robot_ID == 0) ref.read();
+    Serial.println("Ref system online");
+
+
+    // check SD
+    if (sdcard.exists("/config.pack")) {
```

Comment:
put this filepath as a variable (const char*) somewhere and reference the variable.